cmd/ipfs-gateway-fs/main.go

```diff
@@ -53,13 +65,19 @@ func run() error {
 			return err
 		}
 	}
-
 	repo := fsrepo.At(repoPath)
 	if err := repo.Open(); err != nil { // owned by node
 		return err
 	}
-
-	node, err := core.NewIPFSNode(ctx, core.Online(repo))
+	s3, err := makeS3Datastore()
+	if err != nil {
+		return err
+	}
+	enhanced, err := enhanceDatastore(s3)
+	if err != nil {
+		return err
+	}
+	node, err := core.NewIPFSNode(ctx, core.OnlineWithRouting(repo, corerouting.GrandCentralServer(enhanced)))
```

Comment:
The datastore passed to GCR is used _only_ for routing records. The node still has a local datastore.

added comment daaee56a78eadb8da742455ff1ab384dca89d46b